corehq/mobile_ucr_v2_update_script.py

```diff
@@ -0,0 +1,110 @@
+# This script updates the latest versions of all apps across domains that are using v1 with no manual references
+# Steps followed
+# 1. Get All domains with mobile ucr flag enabled
+# 2. Get all apps for domain with latest released versions and mobile ucr versions that are not v2
+# 3. For each app, if it contains no V1 UCR references, update the version to 2
+
+# How to run
+# Can be run in django shell. Paste the script and execute the function process()
+# File is stored in home directory of cchq user.
+
+# V1 Examples
+# https://staging.commcarehq.org/a/test-commcare-superset/apps/view/f940fcc83bae44b8a0adaf719673fd1e/form/a0f3c5b483c645e78b6f89ee0b3b3c03/source/#form/table_child_count
+
+
+import json
+import re
+import traceback
+
+from corehq.apps.app_manager.dbaccessors import (
+    get_latest_app_ids_and_versions,
+    get_apps_by_id,
+)
+from corehq.apps.app_manager.const import MOBILE_UCR_VERSION_2
+from corehq.toggles import MOBILE_UCR
+from corehq.toggles.shortcuts import find_domains_with_toggle_enabled
+from corehq.util.log import with_progress_bar
+
+
+PROCESSED_DOMAINS_PATH = '/home/zandre/cchq/updated_domains.ndjson'
+LOG_FILE = '/home/zandre/cchq/update_to_v2_ucr_script.log'
+
```

Comment:
Hey @zandre-eng , based on the [comment here](https://github.com/dimagi/commcare-hq/pull/35168#discussion_r1800689494) in the other PR, requesting changes to cover the Case List/Detail scenario

```suggestion

# Pattern for references in case list/detail. 
# The below instance needs to be referred in a fixture setup and is then later used in Case List/Detail filters and columns.
V1_CASE_LIST_REFERENCES_PATTERN = r"instance\('commcare:reports'\)/reports/report\[@id='.*'\]/rows/row"
RE_V1_CASE_LIST_REFERENCES_PATTERN = re.compile(V1_CASE_LIST_REFERENCES_PATTERN)
```

Reply:
Thanks for flagging this, I have updated the script to also check for case list references in [6dcc719](https://github.com/dimagi/commcare-hq/pull/35158/commits/6dcc7195eb8f0401f068c92531e9c24a3f9ba031).